hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/RewardCalculator.java

```diff
@@ -0,0 +1,85 @@
+package com.hedera.services.ledger.accounts.staking;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.state.merkle.MerkleStakingInfo;
+import com.hedera.services.utils.EntityNum;
+import com.swirlds.merkle.map.MerkleMap;
+
+import javax.inject.Inject;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.function.Supplier;
+
+public class RewardCalculator {
+	public static final EntityNum stakingFundAccount = EntityNum.fromLong(800L);
+	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts;
+	private final Supplier<MerkleMap<EntityNum, MerkleStakingInfo>> stakingInfo;
+
+	public static final ZoneId zoneUTC = ZoneId.of("UTC");
+
+	@Inject
+	public RewardCalculator(final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts,
+			final Supplier<MerkleMap<EntityNum, MerkleStakingInfo>> stakingInfo) {
+		this.accounts = accounts;
+		this.stakingInfo = stakingInfo;
+	}
+
+	public final long computeAndApplyRewards(final EntityNum accountNum) {
+		long todayNumber = LocalDate.now(zoneUTC).toEpochDay();
```

Comment:
This class will keep evolving; but in all cases, imo we will want to encapsulate the logic that computes the current `stakingPeriod` in a separate `StakingPeriodManager`.

(That logic will be quite involved, with probably a special case to use calendar day periods...would be good to have it in one place.)

Reply:
Makes sense . Will make the change in my current branch.